### PR TITLE
Twitter to Signal

### PR DESCRIPTION
**Pull request by AetherUnbound**
This PR adds some new functionality to the scanner bot in the form of a Twitter-to-Signal transport.

When the bot is launched, a secondary task will "track" (Twitter API term) the desired hashtags. If the "auto-scanner" is enabled, tweets that meet certain criteria (namely from trusted tweeters) will be sent out to the Signal channel. The "auto-scanner" can be enabled or disabled by anyone in the Signal group by messaging "Auto scan on" or "Auto scan off" (casing and whitespace don't matter).

A couple of technical changes had to be made in order to accomplish this:
* I renamed a bunch of functions/constants and moved things around so that dependencies could be managed a bit better and things could be clearer
* It turns out that the `signal-cli` client cannot run more than one instance at a time. Since both tasks are running concurrently, this caused a deadlock since the signal listener was always running. In order to manage this, I've changed the signal listening loop to stop periodically (every 10 seconds) to allow other `signal-cli` calls to run. Additionally, I've added a python lock to make sure that no deadlocks occur.
* Tweepy has an "async" option for their streaming API, but it runs the stream in a separate thread, not python's `async` interface. So I had to create a task by using `asyncio`'s interaction with `ThreadPoolExecutor`.

All that to say, I've got this out in production now and it appears to be working well!

### CODE REVIEW

**Review by JacksonMaxfield — APPROVED** [review]
This all _looks_ good to me. It is somewhat hard to follow but I think that is probably because I have never _worked_ with the repo itself.

I would ask that you probably include the PR comment or a generalized form of it into the repo itself. Maybe at the top of `bin/run.py`?

I think it would just be good to have that as like "hey this loop will run until requested interrupt which then triggers this other script to run for 10 seconds which then exits and the entire system restarts w/ the original loop triggered" -- if that is what happens then I understood you comment a bit more than I think I did which is cool.

**Comment by JacksonMaxfield** [comment]
I am also surprised you couldn't just run two processes and maybe write a temp file / files to a temp dir that are picked up by the secondary process, processed, then the picked up file removed. That seems like it would solve the locking issue maybe? Not sure just thinking out loud.

**Comment by AetherUnbound** [comment]
> I think it would just be good to have that as like "hey this loop will run until requested interrupt which then triggers this other script to run for 10 seconds which then exits and the entire system restarts w/ the original loop triggered" -- if that is what happens then I understood you comment a bit more than I think I did which is cool.

You're totally right, this could use a more accurate description of what's happening! I'll add that, but TL;DR is that only 1 instance of `signal-cli` can be running at once, and it can either be listening or sending but not both. So these loops go back and forth both listening to their streams, and then sending messages as needed (while balancing which one is using the `signal-cli` lock)
